activesupport/lib/active_support/callbacks.rb

```diff
@@ -286,9 +286,8 @@ def self.simple(callback_sequence, user_callback)
       class Callback #:nodoc:#
         def self.build(chain, filter, kind, options)
           if filter.is_a?(String)
-            ActiveSupport::Deprecation.warn(<<-MSG.squish)
-              Passing string to define callback is deprecated and will be removed
-              in Rails 5.1 without replacement.
+            raise ArgumentError, <<-MSG.squish
+              Passing string to define callback is not supported. Use proc instead.
             MSG
           end
```

Comment:
In that case we will have to deprecate it now since we didn't before. @kamipo could you also deprecate it in a new commit in this PR?